session.go

```diff
@@ -973,24 +973,71 @@ func (st *StmtTxn) SeekReverse(k kv.Key) (kv.Iterator, error) {
 	return kv.NewUnionIter(bufferIt, retrieverIt, true)
 }
 
-// StmtCommit implements the Transaction interface.
-func (st *StmtTxn) StmtCommit() error {
-	err := kv.WalkMemBuffer(st.buf, func(k kv.Key, v []byte) error {
+// StmtCommit implements the context.Context interface.
+func (s *session) StmtCommit() error {
+	st := &s.StmtTxn
+	err := kv.WalkMemBuffer(s.buf, func(k kv.Key, v []byte) error {
 		if len(v) == 0 {
 			return errors.Trace(st.Transaction.Delete(k))
 		}
 		return errors.Trace(st.Transaction.Set(k, v))
 	})
 	st.buf.Reset()
+
+	// Need to flush binlog.
+	for key, value := range st.mutations {
+		mutation := getBinlogMutation(s, key)
+		if err == nil {
+			mergeToMutation(mutation, value)
+		}
+		delete(st.mutations, key)
```

Comment:
whatever the err, we will need to cleanup the st.mutations, so why not use a function to do cleanup, I think that will be more readable.

Reply:
Isn't it simple enough?
Or, you think this code is more readable:

```
func (s *session) StmtCommit() error {
    defer s.StmtRollback()
    ....
}
```